helm-charts/splunk-otel-collector/templates/operator/_helpers.tpl

```diff
@@ -103,22 +102,28 @@ Helper to define entries for instrumentation libraries.
 {{- end -}}
 
 {{/*
-Helper to convert a list of dictionaries into a list of keys.
-- Iterates through a list of dictionaries and collects the 'name' field from each.
-- Returns a list of these 'name' keys.
+Helper to check if env (list of dictionaries) has an environment variable (dictionary) with a specific name.
+- Takes a list of dictionaries (env) and a string (envName).
+- Returns "true" if a dictionary in the list has a 'name' key that matches envName, otherwise "false".
 */}}
-{{- define "splunk-otel-collector.operator.extract-name-keys-from-dict-list" -}}
-  {{- /* Initialize variables */ -}}
-  {{- $listOfDicts := . -}}
-  {{- $keyList := list -}}
-
-  {{- /* Collect 'name' field from each dictionary */ -}}
-  {{- range $listOfDicts -}}
-    {{- $keyList = append $keyList .name -}}
+{{- define "splunk-otel-collector.operator.env-has" -}}
+  {{- /* Extract parameters */ -}}
+  {{- $env := default list .env -}}
+  {{- $envName := .envName -}}
+  {{- $found := false -}}
+
+  {{- /* Check if envName exists in the list of dictionaries */ -}}
+  {{- range $env -}}
+    {{- if eq .name $envName -}}
+      {{- $found = true -}}
+    {{- end -}}
   {{- end -}}
 
-  {{- /* Return the list of 'name' keys */ -}}
-  {{- $keyList -}}
+  {{- if $found -}}
+    {{- "true" -}}
+  {{- else -}}
+    {{- "false" -}}
+  {{- end -}}
```

Comment:
You don't need this. Just `{{- $found -}}` should work

Reply:
Done